browser-extension/mv3/src/service-worker/services/sessionRecording.ts

```diff
@@ -78,32 +73,102 @@ export const watchRecording = (tabId: number) => {
 };
 
 const startRecording = (tabId: number, config: Record<string, any>) => {
-  injectWebAccessibleScript("libs/requestly-web-sdk.js", {
-    tabId: tabId,
-  }).then(() => {
-    chrome.tabs.sendMessage(tabId, {
-      action: CLIENT_MESSAGES.START_RECORDING,
-      payload: config,
-    });
+  chrome.tabs.sendMessage(tabId, {
+    action: CLIENT_MESSAGES.START_RECORDING,
+    payload: config,
   });
 };
 
 export const stopRecording = (tabId: number, openRecording: boolean) => {
-  chrome.tabs.sendMessage(tabId, { action: CLIENT_MESSAGES.STOP_RECORDING }).then(() => {
-    if (openRecording) {
-      watchRecording(tabId);
-    }
+  chrome.tabs.sendMessage(tabId, { action: CLIENT_MESSAGES.STOP_RECORDING }, () => {
+    tabService.removeData(tabId, TAB_SERVICE_DATA.SESSION_RECORDING);
   });
+
+  if (openRecording) {
+    watchRecording(tabId);
+  }
 };
 
-export const startRecordingExplicitly = async (tab: chrome.tabs.Tab, showWidget: boolean) => {
+export const startRecordingExplicitly = async (tab: chrome.tabs.Tab, showWidget: boolean = true) => {
+  const sessionRecordingConfig = await getSessionRecordingConfig(tab.url);
+
+  const sessionRecordingDataExist = !!tabService.getData(tab.id, TAB_SERVICE_DATA.SESSION_RECORDING);
+  // Auto recording is on for current tab if sessionRecordingConfig exist,
+  // so forcefully start explicit recording.
+  if (!sessionRecordingConfig && sessionRecordingDataExist) {
+    return;
+  }
+
   const sessionRecordingData = { explicit: true, showWidget };
+  tabService.setData(tab.id, TAB_SERVICE_DATA.SESSION_RECORDING, sessionRecordingData);
 
   startRecording(tab.id, sessionRecordingData);
 };
 
 export const launchUrlAndStartRecording = (url: string) => {
   chrome.tabs.create({ url }, (tab) => {
-    startRecording(tab.id, { explicit: true, notify: true });
+    tabService.setData(tab.id, TAB_SERVICE_DATA.SESSION_RECORDING, {
+      notify: true,
+      explicit: true,
+      showWidget: true,
+    });
+  });
+};
+
+export const handleSessionRecordingOnClientPageLoad = async (tab: chrome.tabs.Tab, frameId: number) => {
+  let sessionRecordingData = tabService.getData(tab.id, TAB_SERVICE_DATA.SESSION_RECORDING);
+
+  if (!sessionRecordingData) {
+    const sessionRecordingConfig = await getSessionRecordingConfig(tab.url);
+
+    if (sessionRecordingConfig) {
+      sessionRecordingData = { config: sessionRecordingConfig, url: tab.url };
+      const recordingMode = sessionRecordingConfig?.autoRecording?.mode;
+
+      sessionRecordingData.showWidget = recordingMode === "custom";
+
+      if (recordingMode === "allPages") {
```

Comment:
any enum type defined for recordingMode config?

Reply:
yes we have